x-pack/plugins/secops/public/components/source_destination/source_destination_arrows.tsx

```diff
@@ -0,0 +1,255 @@
+  destinationPackets: string | undefined;
+}>(({ contextId, eventId, destinationBytes, destinationBytesPercent, destinationPackets }) => {
+  const destinationArrowHeight =
+    destinationBytesPercent != null
+      ? getArrowHeightFromPercent(destinationBytesPercent)
+      : DEFAULT_ARROW_HEIGHT;
+
+  return (
+    <EuiFlexGroup alignItems="center" gutterSize="none" justifyContent="center">
+      <EuiFlexItem grow={false}>
+        <ArrowHead direction="arrowLeft" />
+      </EuiFlexItem>
+
+      <EuiFlexItem grow={false}>
+        <ArrowBody height={destinationArrowHeight} />
+      </EuiFlexItem>
+
+      {destinationBytes != null && !isNaN(Number(destinationBytes)) ? (
+        <EuiFlexItem grow={false}>
+          <DefaultDraggable
+            field={DESTINATION_BYTES_FIELD_NAME}
+            id={`${contextId}-${eventId}-${DESTINATION_BYTES_FIELD_NAME}-${destinationBytes}`}
+            value={destinationBytes}
+          >
+            <Data size="xs">
+              {destinationBytesPercent != null ? (
+                <Percent data-test-subj="destination-bytes-percent">
+                  {`(${numeral(destinationBytesPercent).format('0.00')}%)`}
+                </Percent>
+              ) : null}
+              <span data-test-subj="destination-bytes">
+                {numeral(destinationBytes).format('0.000 b')}
+              </span>
+            </Data>
+          </DefaultDraggable>
+        </EuiFlexItem>
+      ) : null}
+
+      <EuiFlexItem grow={false}>
+        <ArrowBody height={destinationArrowHeight} />
+      </EuiFlexItem>
+
+      {destinationPackets != null && !isNaN(Number(destinationPackets)) ? (
+        <EuiFlexItem grow={false}>
+          <DefaultDraggable
+            field={DESTINATION_PACKETS_FIELD_NAME}
+            id={`${contextId}-${eventId}-${DESTINATION_PACKETS_FIELD_NAME}-${destinationPackets}`}
+            value={destinationPackets}
+          >
+            <Data size="xs">
+              <span data-test-subj="destination-packets">{`${numeral(destinationPackets).format(
+                '0,0'
+              )} ${i18n.PACKETS}`}</span>
+            </Data>
+          </DefaultDraggable>
+        </EuiFlexItem>
+      ) : null}
+
+      <EuiFlexItem grow={false}>
+        <ArrowBody height={destinationArrowHeight} />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+});
+
+/**
+ * Visualizes the communication between a source and a destination using arrows
+ * that grow in thickness based on the percentage of bytes transferred, and stats badges
+ */
+export const SourceDestinationArrows = pure<{
+  contextId: string;
+  destinationBytes?: string[] | null;
+  destinationPackets?: string[] | null;
+  eventId: string;
+  sourceBytes?: string[] | null;
+  sourcePackets?: string[] | null;
+}>(({ contextId, destinationBytes, destinationPackets, eventId, sourceBytes, sourcePackets }) => {
+  const maybeSourceBytes =
+    sourceBytes != null && hasOneValue(sourceBytes) ? sourceBytes[0] : undefined;
```

Comment:
Your `hasOneValue` handles null, so you should only need:

```
hasOneValue(sourceBytes) ? sourceBytes[0] : undefined;
```

Here and below. That would clean up some of these icky checks not needed and cut down on some of the code noise.